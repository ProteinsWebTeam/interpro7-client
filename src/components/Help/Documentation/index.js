// @flow
import React, { PureComponent } from 'react';

import Link from 'components/generic/Link';
import loadable from 'higherOrder/loadable';
import { schemaProcessDataPageSection } from 'schema_org/processors';

import { foundationPartial } from 'styles/foundation';

import fonts from 'EBI-Icon-fonts/fonts.css';
import ipro from 'styles/interpro-new.css';
import ebiGlobalStyles from 'ebi-framework/css/ebi-global.css';
import local from './style.css';

import { InterProLogo } from 'components/Header/Title';

const f = foundationPartial(ebiGlobalStyles, fonts, ipro, local);

const SchemaOrgData = loadable({
  loader: () => import(/* webpackChunkName: "schemaOrg" */ 'schema_org'),
  loading: () => null,
});
export default class Documentation extends PureComponent /*:: <{}> */ {
  render() {
    return (
      <section>
        <SchemaOrgData
          data={{
            name: 'InterPro Documentation Page',
            description:
              'Includes links to the documentations for the different parts of our service',
          }}
          processData={schemaProcessDataPageSection}
        />
        <div className={f('row')}>
          <div className={f('columns', 'large-8')}>
            <div className={f('flex-container')}>
              <div className={f('flex-box')}>
                <InterProLogo /> <h5>Interpro</h5>
                <li>
                  <ul>
                    <li>
                      <Link
                        to={{
                          description: {
                            other: ['about', 'interpro'],
                          },
                        }}
                      >
                        About InterPro
                      </Link>
                    </li>
                    <li>
                      <Link
                        to={{
                          description: {
                            other: ['release_notes'],
                          },
                        }}
                      >
                        Release Notes
                      </Link>
                    </li>
                    <li>
                      <Link
                        to={{
                          description: {
                            other: ['help', 'faqs'],
                          },
                        }}
                      >
                        Frequently Asked Questions (FAQs)
                      </Link>
                    </li>
                    <li>
                      <Link
                        to={{
                          description: {
                            other: ['about', 'ida'],
                          },
                        }}
                      >
                        InterPro Domain Architecture
                      </Link>
                    </li>
                  </ul>
                </li>
              </div>
              <div className={f('flex-box')}>
                <span
                  className={f('icon', 'icon-common', 'xl')}
                  data-icon="&#xf2fd;"
                />{' '}
                <h5>InterProScan</h5>
                <ul>
                  <li>
                    <Link
                      to={{
                        description: {
                          other: ['about', 'interproscan'],
                        },
                      }}
                    >
                      About InterProScan
                    </Link>
                  </li>
                  <li>
                    <Link
                      href="//github.com/ebi-pf-team/interproscan/wiki"
                      className={f('ext')}
                      target="_blank"
                    >
                      <span
                        className={f('icon', 'icon-common')}
                        data-icon="&#xf09b;"
                      />{' '}
                      InterProScan Wiki
                    </Link>
                  </li>
                </ul>
              </div>
              <div className={f('flex-box')}>
                <span
                  className={f('icon', 'icon-common', 'xl')}
                  data-icon="&#xf085;"
                />
                <h5>Technical specs</h5>
                <ul>
                  <li>
                    <Link
                      href="https://github.com/ProteinsWebTeam/interpro7-api/tree/master/docs"
                      className={f('ext')}
                      target="_blank"
                    >
                      <span
                        className={f('icon', 'icon-common')}
                        data-icon="&#xf09b;"
                      />{' '}
                      General API documentation
                    </Link>
                  </li>
                  <li>
                    <Link
                      href="//www.ebi.ac.uk/interpro/api/static_files/swagger/"
                      className={f('ext')}
                      target="_blank"
                    >
                      Swagger API documentation
                    </Link>
                  </li>
                  <li>
                    <Link
                      to={{
                        description: {
                          main: { key: 'result' },
                          result: { type: 'download' },
                        },
                        hash: '/entry/interpro',
                      }}
                    >
                      Code snippet generator
                    </Link>
                  </li>
                  <li>
                    <Link
                      href="//www.ebi.ac.uk/Tools/webservices/"
                      className={f('ext')}
                      target="_blank"
                    >
                      Web services
                    </Link>
                  </li>
                  <li>
                    <Link
                      href="//www.ebi.ac.uk/data-protection/privacy-notice/embl-ebi-public-website"
                      className={f('ext')}
                      target="_blank"
                    >
                      Privacy and security
                    </Link>
                  </li>

                  <li>
                    <Link
                      href="//www.ebi.ac.uk/ebisearch/documentation.ebi"
                      className={f('ext')}
                      target="_blank"
                    >
                      Search documentation (EBI)
                    </Link>
                  </li>
                </ul>
              </div>
              <div className={f('flex-box')}>
                <span
                  className={f('icon', 'icon-common', 'xl')}
                  data-icon="&#xf02d;"
                />{' '}
                <h5>Publications</h5>
                <ul>
                  <li>
                    <Link
                      to={{
                        description: {
                          other: ['help', 'documentation', 'publication'],
                        },
                      }}
                    >
                      Show all previous InterPro publications
                    </Link>
                  </li>
                </ul>
              </div>
            </div>
          </div>
          <div className={f('columns', 'large-4')}>
            <div className={f('box-add', 'margin-top-xxlarge')}>
              <h3 className={f('light')}>Contact us/Additional help</h3>
              <ul>
                {
                  // <li>
                  //   <Link href="//www.ebi.ac.uk/support/interpro-general-query">
                  //     <span
                  //       className={f('icon', 'icon-common')}
                  //       data-icon="&#x27a;"
                  //     />{' '}
                  //     Chat with an assistant{' '}
                  //   </Link>
                  // </li>
                }
                <li>
                  <Link
                    href="//www.ebi.ac.uk/support/interpro"
                    target="_blank"
                    withReferrer
                  >
                    <span
                      className={f('icon', 'icon-common')}
                      data-icon="&#x6e;"
                    />{' '}
                    Submit a ticket
                  </Link>
                </li>
              </ul>
            </div>
          </div>
        </div>
      </section>
    );
  }
}
