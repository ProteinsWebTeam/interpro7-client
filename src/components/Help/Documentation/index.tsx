// @flow
import React from 'react';

import config from 'config';
import Link from 'components/generic/Link';
import loadable from 'higherOrder/loadable';
import { schemaProcessDataPageSection } from 'schema_org/processors';

import fonts from 'EBI-Icon-fonts/fonts.css';
import local from './style.css';

import cssBinder from 'styles/cssBinder';
const f = cssBinder(fonts, local);

const SchemaOrgData = loadable({
  loader: () => import(/* webpackChunkName: "schemaOrg" */ 'schema_org'),
  loading: () => null,
});

const Documentation = () => {
  return (
    <section>
      <SchemaOrgData
        data={{
          name: 'InterPro Documentation Page',
          description:
            'Includes links to the documentations for the different parts of our service',
        }}
        processData={schemaProcessDataPageSection}
      />
      <div className={f('vf-grid', 'vf-grid__col-3')}>
        <div className={f('vf-grid__col--span-2')}>
          <div className={f('flex-container')}>
            <div className={f('flex-box')}>
              <span
                className={f('icon', 'icon-common', 'xl')}
                data-icon="&#xf02d;"
              />{' '}
              <h5>Documentations</h5>
              <li>
                <ul>
                  <li>
                    <Link
                      href={config.root.readthedocs.href}
                      className={f('ext')}
                      target="_blank"
                    >
                      InterPro documentation
                    </Link>
                  </li>
                  <li>
                    <Link
                      href="https://pfam-docs.readthedocs.io/en/latest/"
                      className={f('ext')}
                      target="_blank"
                    >
                      Pfam documentation
                    </Link>
                  </li>
                </ul>
              </li>
            </div>
            <div className={f('flex-box')}>
              <span
                className={f('icon', 'icon-common', 'xl')}
                data-icon="&#xf2fd;"
              />{' '}
              <h5>InterProScan</h5>
              <ul>
                <li>
                  <Link
                    href="https://interproscan-docs.readthedocs.io/"
                    className={f('ext')}
                    target="_blank"
                  >
                    InterProScan documentation
                  </Link>
                </li>
                <li>
                  <Link
                    href="//www.ebi.ac.uk/jdispatcher/docs/"
                    className={f('ext')}
                    target="_blank"
                  >
                    Job Dispatcher documentation
                  </Link>
                </li>
              </ul>
            </div>
            <div className={f('flex-box')}>
              <span
                className={f('icon', 'icon-common', 'xl')}
                data-icon="&#xf085;"
              />
              <h5>Technical specs</h5>
              <ul>
                <li>
                  <Link
                    href="https://github.com/ProteinsWebTeam/interpro7-api/tree/master/docs"
                    className={f('ext')}
                    target="_blank"
                  >
                    <span
                      className={f('icon', 'icon-common')}
                      data-icon="&#xf09b;"
                    />{' '}
                    General API documentation
                  </Link>
                </li>
                <li>
                  <Link
                    href="//www.ebi.ac.uk/interpro/api/static_files/swagger/"
                    className={f('ext')}
                    target="_blank"
                  >
                    Swagger API documentation
                  </Link>
                </li>
                <li>
                  <Link
                    to={{
                      description: {
                        main: { key: 'result' },
                        result: { type: 'download' },
                      },
                      hash: '/entry/interpro/|json',
                    }}
                  >
                    Code snippet generator
                  </Link>
                </li>
                <li>
                  <Link
                    href="//www.ebi.ac.uk/data-protection/privacy-notice/embl-ebi-public-website"
                    className={f('ext')}
                    target="_blank"
                  >
                    Privacy and security
                  </Link>
                </li>
                <li>
                  <Link
                    href="//www.ebi.ac.uk/ebisearch/documentation"
                    className={f('ext')}
                    target="_blank"
                  >
                    EBI Search documentation
                  </Link>
                </li>
              </ul>
            </div>
            <div className={f('flex-box')}>
              <span
                className={f('icon', 'icon-common', 'xl')}
                data-icon="&#xf10d;"
              />{' '}
              <h5>Publications</h5>
              <ul>
                <li>
                  <Link
                    href={`${config.root.readthedocs.href}citing.html`}
                    className={f('ext')}
                    target="_blank"
                  >
                    Citing InterPro
                  </Link>
                </li>
                <li>
                  <Link
                    href="https://pfam-docs.readthedocs.io/en/latest/citing-pfam.html"
                    className={f('ext')}
                    target="_blank"
                  >
                    Citing Pfam
                  </Link>
                </li>
              </ul>
            </div>
          </div>
        </div>
        <div className={f('vf-grid__col--span-1')}>
          <div className={f('box-add', 'margin-top-xxlarge')}>
            <h3 className={f('light')}>Contact us/Additional help</h3>
            <ul>
              {
                // <li>
                //   <Link href="//www.ebi.ac.uk/support/interpro-general-query">
                //     <span
                //       className={f('icon', 'icon-common')}
                //       data-icon="&#x27a;"
                //     />{' '}
                //     Chat with an assistant{' '}
                //   </Link>
                // </li>
              }
              <li>
                <Link
                  href="//www.ebi.ac.uk/support/interpro"
                  target="_blank"
                  withReferrer
                >
                  <span
                    className={f('icon', 'icon-common')}
                    data-icon="&#x6e;"
                  />{' '}
                  Submit a ticket
                </Link>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </section>
  );
};

export default Documentation;
