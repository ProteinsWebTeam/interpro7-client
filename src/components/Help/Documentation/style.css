@import '../../../styles/colors.css';

.box-add {
  border-left: 4px solid var(--colors-submain);
  padding: 0 1rem;
}
.box-add ul {
  list-style-type: none;
  margin: 0;
  padding: 0;
  line-height: 2;
}

.flex-container {
  display: flex;
  flex-direction: row;
  flex-wrap: nowrap;
  padding: 1rem;
}
.flex-container a {
  color: white;
}
.flex-item {
  display: flex;
  flex-wrap: nowrap;
  flex-direction: column;
}

.icon-box {
  border: 1px solid var(--colors-vlightgray);
  text-align: center;
  border-radius: 4px;
  padding: 2rem;
  width: 50%;
  justify-content: center;

  &:first-of-type {
    margin-right: 1rem;
  }
  & ul {
    list-style-type: none;
    margin: 0;
  }
  & h4::before {
    font-size: 200%;
  }
}
