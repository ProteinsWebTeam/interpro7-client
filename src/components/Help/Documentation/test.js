// @flow
import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import Documentation from '.';

const renderer = new ShallowRenderer();

describe('<Documentation />', () => {
  test('should render', () => {
    renderer.render(<Documentation />);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});
