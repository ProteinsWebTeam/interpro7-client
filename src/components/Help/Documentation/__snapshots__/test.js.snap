// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Documentation /> should render 1`] = `
<section>
  <Memo(Loadable)
    data={
      Object {
        "description": "Includes links to the documentations for the different parts of our service",
        "name": "InterPro Documentation Page",
      }
    }
    processData={[Function]}
  />
  <div
    className="row"
  >
    <div
      className="columns large-8"
    >
      <div
        className="flex-container"
      >
        <div
          className="flex-box"
        >
          <InterProLogo />
           
          <h5>
            Interpro
          </h5>
          <li>
            <ul>
              <li>
                <Memo(Connect(Link))
                  className="ext"
                  href="//interpro-documentation.readthedocs.io/en/latest/"
                  target="_blank"
                >
                  Website documentation
                </Memo(Connect(Link))>
              </li>
            </ul>
          </li>
        </div>
        <div
          className="flex-box"
        >
          <span
            className="icon icon-common xl"
            data-icon=""
          />
           
          <h5>
            InterProScan
          </h5>
          <ul>
            <li>
              <Memo(Connect(Link))
                className="ext"
                href="https://interproscan-docs.readthedocs.io/"
                target="_blank"
              >
                InterProScan Documentation
              </Memo(Connect(Link))>
            </li>
            <li>
              <Memo(Connect(Link))
                className="ext"
                href="//www.ebi.ac.uk/seqdb/confluence/display/JDSAT/InterProScan+5+Help+and+Documentation"
                target="_blank"
              >
                Web services
              </Memo(Connect(Link))>
            </li>
          </ul>
        </div>
        <div
          className="flex-box"
        >
          <span
            className="icon icon-common xl"
            data-icon=""
          />
          <h5>
            Technical specs
          </h5>
          <ul>
            <li>
              <Memo(Connect(Link))
                className="ext"
                href="https://github.com/ProteinsWebTeam/interpro7-api/tree/master/docs"
                target="_blank"
              >
                <span
                  className="icon icon-common"
                  data-icon=""
                />
                 
                General API documentation
              </Memo(Connect(Link))>
            </li>
            <li>
              <Memo(Connect(Link))
                className="ext"
                href="//www.ebi.ac.uk/interpro/api/static_files/swagger/"
                target="_blank"
              >
                Swagger API documentation
              </Memo(Connect(Link))>
            </li>
            <li>
              <Memo(Connect(Link))
                to={
                  Object {
                    "description": Object {
                      "main": Object {
                        "key": "result",
                      },
                      "result": Object {
                        "type": "download",
                      },
                    },
                    "hash": "/entry/interpro",
                  }
                }
              >
                Code snippet generator
              </Memo(Connect(Link))>
            </li>
            <li>
              <Memo(Connect(Link))
                className="ext"
                href="//www.ebi.ac.uk/data-protection/privacy-notice/embl-ebi-public-website"
                target="_blank"
              >
                Privacy and security
              </Memo(Connect(Link))>
            </li>
            <li>
              <Memo(Connect(Link))
                className="ext"
                href="//www.ebi.ac.uk/ebisearch/documentation.ebi"
                target="_blank"
              >
                Search documentation (EBI)
              </Memo(Connect(Link))>
            </li>
          </ul>
        </div>
        <div
          className="flex-box"
        >
          <span
            className="icon icon-common xl"
            data-icon=""
          />
           
          <h5>
            Publications
          </h5>
          <ul>
            <li>
              <Memo(Connect(Link))
                className="ext"
                href="//interpro-documentation.readthedocs.io/en/latest/citing.html"
                target="_blank"
              >
                InterPro Publications
              </Memo(Connect(Link))>
            </li>
          </ul>
        </div>
      </div>
    </div>
    <div
      className="columns large-4"
    >
      <div
        className="box-add margin-top-xxlarge"
      >
        <h3
          className="light"
        >
          Contact us/Additional help
        </h3>
        <ul>
          <li>
            <Memo(Connect(Link))
              href="//www.ebi.ac.uk/support/interpro"
              target="_blank"
              withReferrer={true}
            >
              <span
                className="icon icon-common"
                data-icon="n"
              />
               
              Submit a ticket
            </Memo(Connect(Link))>
          </li>
        </ul>
      </div>
    </div>
  </div>
</section>
`;
